v2/client/receiver_test.go

```diff
@@ -38,6 +39,9 @@ func TestReceiverFnValidTypes(t *testing.T) {
 		"ctx in, Event+Result out":       func(context.Context) (*event.Event, protocol.Result) { return nil, nil },
 		"Event in, Event+Result out":     func(event.Event) (*event.Event, protocol.Result) { return nil, nil },
 		"ctx+Event in, Event+Result out": func(context.Context, event.Event) (*event.Event, protocol.Result) { return nil, nil },
+
+		"input contravariance; may accept supertype": func(event.EventReader) {},
```

Comment:
@n3wscott  Maybe this is too loose/flexible, I mean: maybe we don't want to allow (sub|super) types. That would mean replace `ConvertibleTo()` by simple equals `==`.

Reply:
`event.Event` is a struct, not an interface so I think it has to be an `Event` and not an `EventReader`.

https://github.com/cloudevents/sdk-go/blob/master/v2/event/event.go#L11

Reply:
oooh I see, you are asking for just a subset of `event.Event` and in this case it is the `event.EventReader` interface but you could invent your own...

Reply:
>but you could invent your own
Not sure what you meant by this, AFAIK you cannot implement custom interface for builtin type.

Reply:
But it might be interesting sometimes if you could.

Reply:
Fewer languages allow that (e.g. https://clojure.org/reference/protocols).